### DISCUSSION TITLE
Week3 check in

### DISCUSSION THREAD

**Discussion by NaNa7Miiii** [UW-Upwelling-Project/Summer-2023#7] [General]
# Week 3 Check in

**Comment by eeholmes** [comment]
@NaNa7Miiii @minhphan03 The dhub is back up. My efforts at linking up the shared drive was causing a problem. Sorry I didn't notice earlier.

**Comment by minhphan03** [comment]
## Monday check-in:
Investigated geopotential data from ERA5 (Copernicus) with quicker download time than ERA5. Geopotential data only consists measurements at mean sea level, which is arguably what we want. 

ERDDAP data for u and v wind data took around 1.5 hrs each to download for all 21 years. I also downloaded ERA5 data for u and v wind, and as they have the same grid (0.25 degrees), the can substitute each other (if necessary, ?)

Will work on masking new data and downloading new sets if necessary.

**Comment by NaNa7Miiii** [comment]
### Jiarui Monday Checkin
(1) **Updated pretrained-model infos** (attached them in the **[useful links](https://github.com/UW-Upwelling-Project/Summer-2023/discussions/5)**)
(2) Created two buffers on a new bounding box **(lats = slice(-5, 35), lons = slice(50, 90))**
<img width="528" alt="Screenshot 2023-06-26 at 4 06 16 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/5307680f-f3e6-40c0-b432-111635c58a8a">
<img width="514" alt="Screenshot 2023-06-26 at 4 07 59 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/8b504401-c06e-416f-a3f8-9f62fb62ee7c">
<img width="514" alt="Screenshot 2023-06-26 at 4 08 15 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/f9aba24a-2f02-4738-8820-d76480602520">
**sample image showing connected points if their sst difference exceeds a threshold:**
<img width="640" alt="Screenshot 2023-06-26 at 4 08 55 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/0a617a4b-2b4f-4f8a-b77f-f357ee900180">

**Reply by eeholmes** [reply]
https://www.researchgate.net/publication/350519578_Observational_evidence_of_stratification_control_of_upwelling_and_pelagic_fishery_in_the_eastern_Arabian_Sea/figures?lo=1

**Reply by eeholmes** [reply]
<img width="991" alt="image" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/2545978/4c7063ea-9549-48a0-8afe-6d2213d0e14a">

**Reply by eeholmes** [reply]
Here is the erddap fig for 9-01-2021
http://apdrc.soest.hawaii.edu/erddap/griddap/hawaii_soest_d27d_2672_0aac.graph?sst[(2021-09-01T03:00:00Z)][(0.0)][(5.0):(30.0)][(40.0):(90.0)]&.draw=surface&.vars=longitude%7Clatitude%7Csst&.colorBar=%7C%7C%7C%7C%7C&.bgColor=0xffccccff

**Reply by eeholmes** [reply]
<img width="374" alt="image" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/2545978/90a6e721-29a4-49eb-91e2-110856e5d22e">

**Reply by NaNa7Miiii** [reply]
![Uploading Screenshot 2023-06-26 at 5.52.43 PM.png…]()

**Comment by minhphan03** [comment]
# Minh's Notes on geopotential

## Geopotential, introduced
* Geopotential calculation:
![image](https://github.com/UW-Upwelling-Project/Summer-2023/assets/66846878/d4188d30-2c39-487e-8fe6-045be63e2074)
* The geopotential at a particular point in the atmosphere depends only on the height of that point and not on the path through which the unit mass is taken in reaching that point. 
* Geopotential depends on height and gravity, but gravity can vary depending on the height itself (Newton's law of gravity), and longitude and latitude (largest at poles and lowest at equator, due to shape of Earth and variations in the planet's topography).
* Geopotential height is defined as z_g = phi/g_0, where g_0 = 9.80665 m/s^2 (unit is meter (m)). Comparing to the original calculation phi (geopotential)= gz, geopotential height and classic height are only the same if g_0=g. Therefore, if at some point g > g_0, then for the same pressure surface, the geopotential height at that point would be larger than typical.
* Thickness is the geopotential height difference between two pressure levels, in practice most commonly 1000 and 500 hPa. Larger (smaller) 1000–500-hPa thickness indicates: a larger (smaller) distance between the 1000- and 500-hPa height surfaces, and warmer (colder) average temperature in the 1000–500-hPa layer. [ScienceDirect](https://www.sciencedirect.com/topics/earth-and-planetary-sciences/geopotential-height)
* Visualized below: cooler areas have thinner "thickness" with more rapid pressure change than warmer areas, causing winds and heat transfer between these two areas (pressure gradient force), and the tilting combined with Coriolis effect to cause the geostrophic winds to instead move along the contours (last image).
![image](https://github.com/UW-Upwelling-Project/Summer-2023/assets/66846878/5901b394-cfa8-49c6-9547-a97f52f1299d)


## Some mention of geopotential and (thermal) wind
[Scroll down to figure 11.24c](https://geo.libretexts.org/Bookshelves/Meteorology_and_Climate_Science/Practical_Meteorology_%28Stull%29/11%3A_General_Circulation/11.05%3A_Section_6-)
![image](https://github.com/UW-Upwelling-Project/Summer-2023/assets/66846878/b9f164d3-631b-46a4-9298-0cd85cf80e1e)
This a weather map of geopotential heights of the 50 kPa isobaric surface. L and H indicate low and high heights. The black arrow at A shows the geostrophic wind (G2), drawn parallel to the height contours with low heights to its left. Wind direction follows along the contours of cold-warm pressure surfaces.

## Geopotential height, geostrophic wind, and pressure surface
* [GMU](http://colaweb.gmu.edu/dev/clim301/lectures/wind/wind-geo)
* [Definition of geostrophic winds/currents](https://en.wikipedia.org/wiki/Geostrophic_wind)
* [Another analogy on geostrphic wind](https://www.hko.gov.hk/en/education/weather/meteorology-basics/00010-geostrophic-wind.html)

**Reply by minhphan03** [reply]
https://en.wikipedia.org/wiki/Geostrophic_current mentions a little bit about geostrophic transport (geostrophic wind AND current), which can explain the process of upwelling in [this paper](https://www.mdpi.com/2073-4441/15/6/1168) through Ekman transport:
* Ekman transport is fundamental in determining the distribution of heat, salt, and nutrients in the ocean and plays a role in ocean circulation patterns.
* In addition, the sea surface height (SSH) and sea level anomaly (SLA) subsequently may also play an important dynamical role in coastal regions through geostrophic transport.
* We can use resources in this paper to help finding datasets for our paper..
* 
* . "... ERA5 is the best dataset for evaluating wind intensity in offshore and flat onshore locations, making it suitable for analyzing seasonal and long-term trends in coastal upwelling"
* "Using the daily data time series, the intra-annual (monthly averaged) variability of wind speed is investigated over the Persian Gulf, and the Empirical Orthogonal Function (EOF) analysis has been employed to study the leading patterns of wind speed."

**Comment by minhphan03** [comment]
ERDDAP NOAA CoastWatch for SLA: https://coastwatch.noaa.gov/cwn/products/sea-level-anomaly-and-geostrophic-currents-multi-mission-global-optimal-interpolation.html

Pros: have daily data (no hourly, but better than monthly), most updated data.
Cons: different grid system (still 0.25x0.25deg, but starts at .125 insteads of .0), We can do interpolation (computationally expensive) or simply shift the grid system by .125 deg if the accuracy error is acceptable.

**Comment by NaNa7Miiii** [comment]
### Jiarui Tuesday Check In
### (1) Automatic-Detection adjustment:
**New Bounding Box: lats = slice(-5, 35), lons = slice(45, 90)**
Sampled points along the 100km buffer after noise removal:
<img width="586" alt="Screenshot 2023-06-27 at 3 03 23 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/d2f3c870-578d-484b-b76e-8f4868eeca91">
Adding functions to eliminate gaps along the 300km buffer:
<img width="747" alt="Screenshot 2023-06-27 at 3 08 02 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/93b789fb-efdf-4ba5-ac1f-99d209bfcb09">
<img width="556" alt="Screenshot 2023-06-27 at 3 06 52 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/89af45b7-775e-48e8-b3b7-49b34e14bc19">
It is hard to simulate the color map in ERDDAP, their default color map is 'rainbow', I tried many color maps and found one from Python's matplotlib which is called '**jet**' could be **an alternative for 'rainbow'**:
`plt.contourf(data.columns, data.index, data.values, cmap='jet', levels=np.arange(290, 310.5, 0.5))`
<img width="705" alt="Screenshot 2023-06-27 at 3 24 31 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/76a5e32f-b497-4c89-8cd9-c951994f4dcb">
This is the 2021-07-01 **ERDDAP Image** for SST:
![hawaii_soest_d27d_2672_0aac](https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/575de7e8-51d9-4f9c-ae05-399bc4ccc15e)
### (2) Supervised Learning Research:
Pretrained Convolutional Neural Network (CNN) models, such as EfficientNetB0, are usually designed for processing image data. The data these models use when training is usually RGB images, which are three-dimensional in shape (height, width, color channels). xarray datasets can store multidimensional data, which is somewhat similar to images. So, if the xarray data can be understood as an image (for example, the sea surface temperature data at each time step can be viewed as a 2D image), then you can convert the xarray data into a form acceptable to the model.
**Preprocessing our data from the ERA5 dataset:**
`sst_data = ds['sea_surface_temperature'].values`
`sst_data = sst_data.reshape(-1, len(ds['lat']), len(ds['lon']), 1)`
Then, assume we already have the labels from the automatic detection, we can then do **train-test split**:
`X_train, X_test, y_train, y_test = train_test_split(sst_data, labels, test_size=0.2, random_state=42)`
Then **normalize**:
`X_train = X_train / 255.0`
`X_test = X_test / 255.0`
**Model Adjustment and Training:** (Take EfficientnetB0 as an example)
```
base_model = EfficientNetB0(weights='imagenet', include_top=False)
x = base_model.output
x = GlobalAveragePooling2D()(x) 
predictions = Dense(1, activation='sigmoid')(x)
model = Model(inputs=base_model.input, outputs=predictions)
# layer freezing
for layer in base_model.layers:
    layer.trainable = False
model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])
model.fit(X_train, y_train, epochs=10, validation_split=0.2)
test_loss, test_acc = model.evaluate(X_test, y_test)
```

This is just an idea, and require experiments on the ERA5 dataset.

**Comment by eeholmes-uw** [comment]
## Eli Wednesday update

I worked with a Microsoft rep this morning and we (he) got the shared file service up and running. You'll see a drive called "shared" any files there can be seen by all of us.

**Comment by eeholmes** [comment]
@minhphan03 Looks like interpolating the grids might not be so hard:

I think you might find success with xarray's `ds.interp` https://docs.xarray.dev/en/stable/user-guide/interpolation.html  I think if you load both datasets in to Xarray, you could interpolate either into the other's grid.



There's a `ds.interp_like()` function which takes an `other` argument, which is the dataset whose grid you want to interpolate to.
https://docs.xarray.dev/en/stable/generated/xarray.Dataset.interp_like.html

**Comment by NaNa7Miiii** [comment]
Jiarui Wed Check-In:
(1) Applied the automatic detection on the ERA5 dataset
(2) Test CNN model training using randomly generated xarray sst data & labels:
<img width="800" alt="Screenshot 2023-06-28 at 3 48 02 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/94c5f625-7132-48e0-b188-e0a200c744ea">
<img width="800" alt="Screenshot 2023-06-28 at 3 48 27 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/14c0f021-fd77-4a01-b174-0104e736ce5a">
For this experiment, the pre-trained CNN model (efficientnetB0) **can take xarray as input**.
(3) If we want to predict upwelling, instead of doing image classification that tells whether a day's data indicates upwelling or not, we may want to use other models other than CNN. **Recurrent Neural Networks** (RNNs) would be a better choice compared to CNNs when dealing with time-series data. Because RNNs can remember and process sequential relationships between inputs. In other words, RNNs are able to save states between time steps, while CNNs can only save states between neurons in the same layer. This means that RNNs are able to model temporal dependencies in the input data, which is necessary to predict future events. The other possible choice can be **Long Short Term Memory** Models (LSTM), this paper provides insights for using LSTM models:https://arxiv.org/abs/1506.04214

**Comment by eeholmes** [comment]
## Minh

Copernicus variables
https://cds.climate.copernicus.eu/cdsapp#!/dataset/satellite-sea-level-global?tab=overview

* sea surface height anomaly
* error in that
* geostrophic velocity anomalies - u component
* geostrophic velocity anomalies - v component
* errors in those
* absolute dynamic topography = sla + mean dynamic topography

Sea Level Data
* nice summary of sea level data https://www.aviso.altimetry.fr/en/data/products/ocean-indicators-products/mean-sea-level.html
* 1992 to 2009 reconstructed https://podaac.jpl.nasa.gov/dataset/RECON_SEA_LEVEL_OST_L4_V1
* reanalyzed https://cds.climate.copernicus.eu/cdsapp#!/dataset/satellite-sea-level-global?tab=overview

From paper

* daily wind speed data by a spatial resolution of 0.125° × 0.125°, have been obtained from ERA5 dataset over 28 years
* vertical temperature (air)
* mean SLA from the Copernicus Marine Environment Monitoring Service (CMEMS) dataset with the spatial resolution of 0.25° × 0.25° have been obtained over 28 years (1993–2020).
* mixed layer depth over coasts of the Persian Gulf in the grids with a spatial resolution of 0.25°
 × 0.25° from ARMOR-3D dataset
* geopotential strophic transport - wind generation;

**Comment by eeholmes** [comment]
## Eli Thursday

* blog on creating Zarr files https://medium.com/informatics-lab/create-zarr-from-pp-files-ffa6b7972d6f
* https://measurespace.medium.com/use-zarr-to-access-clound-storage-just-like-your-local-file-system-d67607cb128b
* create Zarr files (notebook) https://aws.amazon.com/blogs/publicsector/decrease-geospatial-query-latency-minutes-seconds-using-zarr-amazon-s3/

I worked on getting to know the Zarr package using the surface_currents.zarr file that Minh put in the shared directory. My work is in eli-notesbooks/zarr-test.ipynb

Reading Zarr from R

* https://www.r-bloggers.com/2022/09/reading-zarr-files-with-r-package-stars/

@NaNa7Miiii Sample code to get xarray from a zarr file. My idea is to create a zarr directory like the surface_currents.zarr file and then we load xarrays from that via a function that filters on a lat/lon bounding box. file: `shared/eli-notebooks/zarr-test.ipynb`

```
import xarray as xr
import numpy as np
zarr_ds = xr.open_zarr(store='/home/jovyan/shared/data/surface_currents.zarr', consolidated=True)
# slice did not work since the lat and lon vals are not indices
latvals = zarr_ds['lat'].values
lonvals = zarr_ds['lon'].values
lats = np.where(np.logical_and(latvals>=10, latvals<=15))[0]
lons = np.where(np.logical_and(lonvals>=60, lonvals<=70))[0]
#time_period = pd.date_range('2019-01-01', '2019-01-31')
time_period = '2019-01-01'
ds = zarr_ds.sel(time=time_period,latitude=lats, longitude=lons)
```

time periods would not be selected based on date, but based on your upwelling indicators (so dates_with_upwelling_sw, dates_with_upwelling_somalia, date_with_upwelling_arabia)

**Reply by eeholmes** [reply]
# to do

* Make a Zarr file
* Make a little function to subset

**Comment by NaNa7Miiii** [comment]
### Jiarui Thursday Check In
Since our goal is to detect where upwelling happened in our bounding box and how many upwellings occur, which means this isn't a yes/no question, so conventional image classification model will not meet our demand here. In order to get image segmentation, I will use **U-Net**. U-Net ([paper](https://arxiv.org/abs/1505.04597)) was developed for biomedical image segmentation, which would be similar to our task.  Here is their demo: https://lmb.informatik.uni-freiburg.de/people/ronneber/u-net/.

I have tested U-Net on some sample tasks today:
### (1) Predict the original image of a noisy image
Model Structure: 
```
def unet(input_size=(64,64,1)):
    inputs = Input(input_size)
    conv1 = Conv2D(32, (3, 3), activation='relu', padding='same')(inputs)
    pool1 = MaxPooling2D(pool_size=(2, 2))(conv1)
    conv2 = Conv2D(64, (3, 3), activation='relu', padding='same')(pool1)
    pool2 = MaxPooling2D(pool_size=(2, 2))(conv2)
    conv3 = Conv2D(128, (3, 3), activation='relu', padding='same')(pool2)
    up4 = concatenate([UpSampling2D(size=(2, 2))(conv3), conv2], axis=-1)
    conv4 = Conv2D(64, (3, 3), activation='relu', padding='same')(up4)
    up5 = concatenate([UpSampling2D(size=(2, 2))(conv4), conv1], axis=-1)
    conv5 = Conv2D(32, (3, 3), activation='relu', padding='same')(up5)
    conv6 = Conv2D(1, (1, 1), activation='sigmoid')(conv5)
    model = Model(inputs=inputs, outputs=conv6)
    model.compile(optimizer=Adam(lr=1e-4), loss='mse')
    return model
```
Result (Original / Noisy / Predicted):
<img width="959" alt="Screenshot 2023-06-29 at 3 05 37 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/9f300e7b-e5b3-4284-8fb8-adb88830ad58">

### (2) Upwelling Detection for Randomly Generated SST Values
In order to use U-Net for our upwelling detection task, we have to make acceptable inputs and labels. 
Here is an example of using U-Net on randomly generated data: 
<img width="959" alt="Screenshot 2023-06-29 at 3 12 04 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/deb935be-3fde-437f-9478-784c4d093249">
The output will be an image that **every pixel has a value between 0 and 1**, which stands for the likelihood of upwelling. We can set a possibility threshold, and for pixels that have values exceeding that threshold we conclude upwelling happens there. For example, in the predicted image, a pixel with a value of 0.7 represents a 70% probability of upwelling in the region corresponding to the sst image. We can set a threshold according to actual needs, and convert the probability into a judgment of occurrence or non-occurrence of upwelling.

**Comment by minhphan03** [comment]
# Minh's Report
* (Almost) complete downloading/subsetting copernicus SHA data [script runs fine, but the server eventually times out so I need to rerun it]
* Found the same dataset with spatial slicing & variable selection options using API request (motuclient) which I'm almost done developing the script to work with this.
https://data.marine.copernicus.eu/product/SEALEVEL_GLO_PHY_L4_MY_008_047/download?dataset=cmems_obs-sl_glo_phy-ssh_my_allsat-l4-duacs-0.25deg_P1D
* Two methods of interpolation: nearest (shift) and linear (default). 
* Thinking of using bash script to handle checking library installs, handle variables (

**Reply by eeholmes** [reply]
# Bounding Box to use

latitude: -12deg to 32deg
longitude: 42deg to 102deg
0.25 deg grid
centers on 0, 0.25, 0.5., 0.75

ERA5
* u wind
* v wind
* wind speed
* wind direction
* air temperature @ 2m
* sea surface temperature

Copernicus
* sea level anomaly
* chlorophyll concentration

**Comment by minhphan03** [comment]
# Update: Friday
* Downloaded all variables -- everything completed
* Having issues trying to save processed data to disk (zarr & netcdf4): issues also happened when trying to save using load() function https://github.com/pydata/xarray/issues/2912
* Interpolation: pending...

**Reply by eeholmes** [reply]
Some ideas

* https://discourse.pangeo.io/t/netcdf-to-zarr-best-practices/1119
* https://speakerdeck.com/rabernat/pangeo-zarr-cloud-data-storage?slide=14
* https://nbviewer.org/gist/TomAugspurger/b4f95f5432fced544d1f23e18ea8368f
* https://stackoverflow.com/questions/73875277/how-to-create-zarr-array-from-a-large-stack-of-npy-files

```
z = zarr.open(
    gcsfs.GCSFileSystem(project=<project-name>).get_mapper(<bucket-name>),
    mode='w',
    shape=(240,256,1440,3),
    dtype=np.float32
)

for i, fn in enumerate(filenames):
    z[i] = np.load(fn)
```

**Reply by eeholmes** [reply]
Data

* ERA5: Daily means
* Copernicus: new bounding box

**Reply by minhphan03** [reply]
Hi @eeholmes sorry for the shutdown my laptop battery died and I was planning to warn you that but it died before that so I’m sorry. I’ll work on the resampling and combining things and I’ll see you next week. Have a great weekend x

**Comment by NaNa7Miiii** [comment]
### Jiarui Friday Check In
Test training using the U-Net model:
First, I stored labels and sst_data with four dimensions: time, lat, lon, value channel
This can be a visualization for labels and sst_data for day 0:
<img width="558" alt="Screenshot 2023-06-30 at 3 54 26 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/605a50d7-be4d-46b7-afe9-97d4abb8a0cd">
<img width="581" alt="Screenshot 2023-06-30 at 3 54 41 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/6ec23deb-a72e-456a-a0bb-a268cdd13dd7">
Since we have to let the dimensions be consistent with the required formats for the U-Net Model, I used padding to enlarge the input sizes:
```
def pad_image_for_unet(img, target_size=(192, 192)):
    pad_before_y = (target_size[0] - img.shape[0]) // 2
    pad_after_y = target_size[0] - img.shape[0] - pad_before_y
    pad_before_x = (target_size[1] - img.shape[1]) // 2
    pad_after_x = target_size[1] - img.shape[1] - pad_before_x

    img_padded = np.pad(img, ((pad_before_y, pad_after_y), 
                              (pad_before_x, pad_after_x), 
                              (0, 0)), 
                        mode='constant', 
                        constant_values=0)

    return img_padded
sst_data_padded = np.array([pad_image_for_unet(img) for img in sst_data])
labels_padded = np.array([pad_image_for_unet(img) for img in labels])
```
Then I did the training, with batch size = 32 and epochs = 350:
```
X_train, X_val, y_train, y_val = train_test_split(sst_data_padded, labels_padded, test_size=0.2, random_state=42)
X_train = np.nan_to_num(X_train)
X_val = np.nan_to_num(X_val)
input_shape = X_train.shape[1:]
model = Unet('resnet34', input_shape=input_shape, encoder_weights=None)
model.compile(optimizer=Adam(), loss=binary_crossentropy, metrics=[iou_score])
history = model.fit(
    x=X_train,
    y=y_train,
    batch_size=32,
    epochs=350,
    validation_data=(X_val, y_val),
)
```
This is the result after training:
`1/1 [==============================] - 4s 4s/step - loss: 0.0025 - iou_score: 0.2001 - val_loss: 0.0055 - val_iou_score: 0.0289`

**Reply by NaNa7Miiii** [reply]
Since my input is just 2017-06's data (just a month), the result seems weak, but I think the accuracy has potential to grow

**Comment by eeholmes** [comment]
# More data to add to list

* Bathymetry https://coastwatch.pfeg.noaa.gov/erddap/griddap/usgsCeSrtm30v6.html
* Probably add MUR data separately since it is finer scale SST. Might be needed for eddy and front detection

**Reply by minhphan03** [reply]
There is no time dimension in the Bathymetry data, so I'll just upload an nc file downloaded from the site for our bounding box (interpolated of course).

**Reply by minhphan03** [reply]
We already have SST data from Copernicus with the same grid as the[ MUR data 0.25x0.25 ](https://search.earthdata.nasa.gov/search/granules?p=C2036880657-POCLOUD!C2207986708-ORNL_CLOUD&pg[1][v]=t&pg[1][gsk]=-start_date&pg[1][m]=download&q=MUR25-JPL-L4-GLOB-v04.2&tl=1686722489!3!!)unless there's another version of it I don't know about? Are we still computing daily data from hourly measurements?
